### PR TITLE
feat: cfn-guard plugin

### PR DESCRIPTION
**Pull request by corymhall**
Initial release of the `cdk-validator-cfnguard` plugin.

See the [README](https://github.com/cdklabs/cdk-validator-cfnguard/blob/corymhall/plugin/README.md) for more details.

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                messages: nestedValue.messages ?? _level2NestedValue.messages ?? value.messages,
+              };
+              break;
+          }
+        });
+      }
+    });
+    return newValue;
+  } else if (key === 'checks' && Array.isArray(value)) {
+    if (value.some(v => Object.values(v).some((vv: any) => {
```
[the remark this answers is not in the document]
Comment (corymhall) [reply src/plugin.ts]: Ok I refactored it into a couple of functions. let me know what you think.

File: src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+ */
+function extractNestedChecks(checks: any[]): any[] {
+  if (checks.some(check => Object.values(check).some((value: any) => {
+    if (typeof value === 'object') {
+      if (value.hasOwnProperty('checks')) {
+        return true;
+      }
+    }
+    return false;
+  }))) {
```
Comment (otaviomacedo) [inline src/plugin.ts:260]:
```suggestion
const containsNestedChecks = checks.some(check => Object.values(check).some((value: any) => {
  return typeof value === 'object' && value.hasOwnProperty('checks');
}));

if (containsNestedChecks) {
```

File: src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+/**
+ * Extract a nested object and pull it up a level
+ */
+function extractNestedObject(object: any): any {
+  let newObject = object;
+  Object.entries(object).forEach(([level1NestedKey, level1NestedValue]) => {
+    const nestedValue = level1NestedValue as any;
+    switch (level1NestedKey.toLowerCase()) {
+      case 'unresolved':
```
Comment (otaviomacedo) [inline src/plugin.ts:289]: It would be helpful to have a small comment for each of these cases explaining what they mean.

Comment (corymhall) [reply src/plugin.ts:308]: Ok added some comments. Let me know if those are good.

**Review by otaviomacedo — APPROVED** [review]

**Review by eppeters — APPROVED** [review]

**Pull request merged**